sparkmagic/sparkmagic/livyclientlib/reliablehttpclient.py

```diff
@@ -61,15 +61,37 @@ def _send_request_helper(self, url, accepted_status_codes, function, data, retry
             try:
                 if self._endpoint.auth == constants.NO_AUTH:
                     if data is None:
```

Comment:
This now has a huge amount of duplicate paths and code, which makes me worry about errors. Can it be simplified? E.g.

```
if data is not None:
    data = json.dumps(data)
```

at the top would simplify it quite a lot, and so on.